apps/teams/middleware.py

```diff
@@ -28,5 +29,11 @@ def _get_team_membership(request):
 
 class TeamsMiddleware(MiddlewareMixin):
     def process_view(self, request, view_func, view_args, view_kwargs):
-        request.team = SimpleLazyObject(lambda: _get_team(request, view_kwargs))
+        try:
+            request.team = _get_team(request, view_kwargs)
+        except Http404:
+            request.team = None
+            request.team_membership = None
+            return HttpResponseNotFound()
```

Comment:
Hmm, this is interesting. What does it show if not the 404 page? Returning `HttpResponseNotFound` puts the onus on us to render the HTML

Reply:
Good point. Alternative approach is to return `None` for the team and let other code like `login_and_team_required` handle the it: [246a50d](https://github.com/dimagi/open-chat-studio/pull/114/commits/246a50d2b22d8420993aff415fcba0494169251d)